### PR TITLE
feat(sue): improve report form

### PR DESCRIPTION
**Pull request by ardasnturk**
- removed the * icon in inputs that are not required to be filled in when sending a report
- removed checks for non-required inputs when submitting the report
- added error message to warn the user if an image with a size higher than 10MB is selected
- removed the `errorMessage` prop from the hidden `Input` component in `ImageSelector` to prevent the error message from being repeated and appearing above the image selection button
- added `RegularText` component to `ImageSelector` to show error message
- added new error message text to `texts.js`
- added a new section for Sue to `dataErrorMessageGenerator.js` to check if the selected image is larger than 10MB and generate an error message
- added controls to `SueReportScreen` to check each step on the report screen and prevent the user from moving to the next step without filling in the missing data
- added a check to check if the selected images are larger than 30MB in total and added the size information of the images to `imagesAttributes`
- added new texts to `texts.js` for error messages
- edited to have a maximum of 5 digits for postcode input
- added an alert to notify if the user types a value different from the 5-digit value in the postcode and prevented the user from proceeding to the next step
- deleted `keyboardType` of `homeNumber` input

SUE-7

## Screenshots:

|non-required input (step 2)|non-required inputs (step 4)|
|--|--|
Simulator Screenshot - iPhone 14 Pro - 2023-12-06 at 16 12 30|Simulator Screenshot - iPhone 14 Pro - 2023-12-06 at 16 12 46

## Alerts:

|step 1 alert|step 2 alert (title)|selection of images larger than 10MB|step 2 alert (Image size higher than 30MB)|
|--|--|--|--|
Simulator Screenshot - iPhone 14 Pro - 2023-12-06 at 16 11 41|Simulator Screenshot - iPhone 14 Pro - 2023-12-06 at 16 11 45|Simulator Screenshot - iPhone 14 Pro - 2023-12-06 at 16 11 55|Simulator Screenshot - iPhone 14 Pro - 2023-12-06 at 16 11 58

|step 3 alert (location not selected from map)|step 3 alert (address information not added)|step 4 alert (user agreement not approved)|
|--|--|--|
Simulator Screenshot - iPhone 14 Pro - 2023-12-06 at 16 12 04|Simulator Screenshot - iPhone 14 Pro - 2023-12-06 at 16 12 09|Simulator Screenshot - iPhone 14 Pro - 2023-12-06 at 16 12 20

### CODE REVIEW

File: src/config/texts.js [hunk 1]
```diff
@@ -914,10 +914,13 @@ export const texts = {
       alerts: {
         address: 'Bitte stellen Sie sicher, dass Sie Ihre Adressdaten korrekt eingeben',
         contact: 'Bitte stellen Sie sicher, dass Sie Ihre Kontaktdaten korrekt eingeben.',
-        email:
-          'Bitte geben Sie eine E-Mail-Adresse an. Ohne Ihre E-Mail-Adresse können wir nicht für eventuelle Rückfrage mit Ihnen Kontakt aufnehmen.',
         hint: 'Hinweis',
-        termsOfService: 'Bitte akzeptieren Sie die Datenschutzbestimmungen.'
+        imageGreater10MBError: 'Das ausgewählte Bild darf maximal 10 MB groß sein.',
+        imagesGreater30MBError: 'Die ausgewählten Bilder dürfen nicht größer als 30 MB sein.',
```
Comment (donni106) [inline src/config/texts.js:919]: --> Die ausgewählten Bilder dürfen insgesamt nicht größer als 30 MB sein.

Comment (ardasnturk) [reply src/config/texts.js]: https://github.com/smart-village-solutions/smart-village-app-app/commit/60e872c4943e1914d253aba8a4ddb45a3bad1599

File: src/screens/SUE/SueReportScreen.tsx [hunk 2]
```diff
@@ -155,7 +138,56 @@ export const SueReportScreen = ({ navigation }: { navigation: any }) => {
       .finally(() => setIsLoading(false));
   };
 
+  /* eslint-disable complexity */
+  const alertTextGeneratorForMissingData = () => {
+    switch (currentPage) {
```
Comment (donni106) [inline src/screens/SUE/SueReportScreen.tsx:143]: `currentProgress` please as used in the component instead of a different name here for the same value.

**Review by donni106 — APPROVED** [review]

File: src/screens/SUE/SueReportScreen.tsx (hunk 2 above)
Comment (ardasnturk) [reply src/screens/SUE/SueReportScreen.tsx]: https://github.com/smart-village-solutions/smart-village-app-app/pull/978/commits/49b11db13461321ba143f833d4bb3ad49e52ca8d

**Pull request merged**